asapdiscovery-ml/asapdiscovery/ml/weights/weights.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from pathlib import Path
+from typing import Dict, List, Union  # noqa: F401
+
+import requests
+import validators
+import yaml
+
+
+def fetch_weights_from_spec(
+    yamlfile: str,
+    models: Union[list[str], str],
+    local_dir: str = "./_weights/",
+    force_fetch: bool = False,
+) -> dict[str, Path]:
+    """Fetch weights from yaml spec file.
+
+    Parameters
+    ----------
+    yamlfile : str
+        Path to yaml spec file.
+    models : List[str]
+        Model names to fetch weights for.
+    local_dir : str, default="./_weights/"
+        Local path to save weights if a remote url is provided. or to check if weights exist locally, by default "./_weights/"
+    force_fetch : bool, default=False
+        Force fetch weights from remote, by default False
+
+    Raises
+    ------
+    FileNotFoundError
+        If YAML spec file does not exist.
+
+    Returns
+    -------
+    Dict of model names and paths to weights files.
+    """
+    if not Path(yamlfile).exists():
+        raise FileNotFoundError(f"Yaml spec file {yamlfile} does not exist")
+
+    with open(yamlfile) as f:
+        spec = yaml.safe_load(f)
+
+    if isinstance(models, str):
+        models = [models]
+
+    weights_files = {}
+    filename_set = set()
+    for model in models:
+        model_spec = spec[model]
+        weights = model_spec["weights"]
+        if weights in filename_set:
+            raise ValueError(
+                f"Duplicate file {weights} in spec file. Please specify a unique filename for each model."
+            )
+        weights_files[model] = fetch_weights(weights, local_dir, force_fetch)
+        filename_set.add(weights)
+    return weights_files
+
+
+def fetch_weights(
+    weights: str, file_dir: str = "./_weights/", force_fetch: bool = False
+) -> Path:
+    """Fetch weights from remote and save to local path.
+
+    Parameters
+    ----------
+    weights : str
+        Remote url or local path to weights.
+    file_dir : str, default="./_weights/"
+        Local path to save weights if a remote url is provided. or to check if weights exist locally, by default None
+    force_fetch : bool, default=False
+        Force fetch weights from remote, by default False
+
+    Returns
+    -------
+    Path to weights file.
+    """
+
+    is_url = validators.url(weights)
+
+    if not file_dir:
+        raise ValueError("file_dir must be provided and must not be falsy.")
```

Comment:
```suggestion
        raise ValueError("file_dir must be provided and must not be False.")
```

Reply:
I think falsy is correct in this case because it will tigger on `None` or any other `False-like`  value, not just false.

Reply:
ah ok, fair enough